kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java

```diff
@@ -281,6 +293,38 @@ private SslContextFactory getSSLContextFactory() {
         return  sslContextFactory;
     }
 
+    /**
+     * Creates a Handler instance to handle incoming HTTP requests for readiness check
+     *
+     * @return Handler
+     */
+    /* test */ Handler getReadinessHandler() {
+        return new AbstractHandler() {
+            @Override
+            public void handle(String s, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException {
+                response.setContentType("application/json");
+                response.setCharacterEncoding("UTF-8");
+                baseRequest.setHandled(true);
+                if (brokerState != null) {
+                    byte observedState = (byte) brokerState.value();
+                    boolean stateIsRunning = BROKER_RUNNING_STATE <= observedState && BROKER_UNKNOWN_STATE != observedState;
+                    if (stateIsRunning) {
+                        LOGGER.trace("Broker is in running according to {}. The current state is {}", brokerStateName, observedState);
+                        response.setStatus(HttpServletResponse.SC_OK);
+                    } else {
+                        LOGGER.trace("Broker is not running according to {}. The current state is {}", brokerStateName, observedState);
+                        response.setStatus(HttpServletResponse.SC_NOT_ACCEPTABLE);
```

Comment:
as per `406 NOT ACCEPTABLE` HTTP status code (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406) I am not sure it's the right use case here.
We are checking a "service" (the broker) which is kind of not available in the state we would like to have ... so maybe `503 SERVICE UNAVAILABLE` is better?
The 4xx status code are client error but here the HTTP client is just asking for the state of the broker, it didn't make any mistake or sent a wrong request. There is nothing the client can do to fix the problem.
It's looks like more a 5xx server error to me.